src/producer/browser/Full.js

```diff
@@ -29,30 +29,66 @@ const FullBrowserProducer = (context) => {
   const splitsUpdater = SplitChangesUpdater(context);
   const segmentsUpdater = MySegmentsUpdater(context);
 
+  let isSynchronizeSplitsRunning = false;
+
+  function synchronizeSplits() {
+    isSynchronizeSplitsRunning = true;
+    return splitsUpdater().finally(function() {
+      isSynchronizeSplitsRunning = false;
+    });
+  }
+
+  let isSynchronizeMySegmentsRunning = false;
+
+  function synchronizeMySegments(segmentList) {
+    isSynchronizeMySegmentsRunning = true;
+    return segmentsUpdater(undefined, segmentList).finally(function () {
+      isSynchronizeMySegmentsRunning = false;
+    });
+  }
+
   const settings = context.get(context.constants.SETTINGS);
   const { splits: splitsEventEmitter } = context.get(context.constants.READINESS);
-  
-  const splitsUpdaterTask = TaskFactory(splitsUpdater, settings.scheduler.featuresRefreshRate);
+
+  const splitsUpdaterTask = TaskFactory(synchronizeSplits, settings.scheduler.featuresRefreshRate);
   const segmentsUpdaterTask = TaskFactory(segmentsUpdater, settings.scheduler.segmentsRefreshRate);
```

Comment:
Shouldn't this be using synchronizeMySegments ? Why only splits use the wrapping function but not the segments?

Reply:
yes!, actually I found a bug, thanks to that comment